tests/components/test_apiai.py

```diff
@@ -0,0 +1,620 @@
+            "timestamp": REQUEST_TIMESTAMP,
+            "result": {
+                "source": "agent",
+                "resolvedQuery": "my zodiac sign is virgo",
+                "speech": "",
+                "action": "unknown",
+                "actionIncomplete": False,
+                "parameters": {
+                    "ZodiacSign": ""
+                },
+                "contexts": [],
+                "metadata": {
+                    "intentId": INTENT_ID,
+                    "webhookUsed": "true",
+                    "webhookForSlotFillingUsed": "false",
+                    "intentName": INTENT_NAME
+                },
+                "fulfillment": {
+                    "speech": "",
+                    "messages": [
+                        {
+                            "type": 0,
+                            "speech": ""
+                        }
+                    ]
+                },
+                "score": 1
+            },
+            "status": {
+                "code": 200,
+                "errorType": "success"
+            },
+            "sessionId": SESSION_ID,
+            "originalRequest": None
+        }
+        req = _intent_req(data)
+        self.assertEqual(200, req.status_code)
+        text = req.json().get("speech")
+        self.assertEqual(
+            "Intent 'unknown' is not yet configured within Home Assistant.",
+            text)
+
+    def test_response_time_async_action(self):
+        """Test response time executing action async. Should not exceed 5s."""
+        data = {
+            "id": REQUEST_ID,
+            "timestamp": REQUEST_TIMESTAMP,
+            "result": {
+                "source": "agent",
+                "resolvedQuery": "too long",
+                "speech": "",
+                "action": "TooLongAsync",
+                "actionIncomplete": False,
+                "parameters": {},
+                "contexts": [],
+                "metadata": {
+                    "intentId": INTENT_ID,
+                    "webhookUsed": "true",
+                    "webhookForSlotFillingUsed": "false",
+                    "intentName": INTENT_NAME
+                },
+                "fulfillment": {
+                    "speech": "",
+                    "messages": [
+                        {
+                            "type": 0,
+                            "speech": ""
+                        }
+                    ]
+                },
+                "score": 1
+            },
+            "status": {
+                "code": 200,
+                "errorType": "success"
+            },
+            "sessionId": SESSION_ID,
+            "originalRequest": None
+        }
+        req_start_time = datetime.now()
+        _intent_req(data)
+        req_end_time = datetime.now()
+        request_time = (req_end_time - req_start_time).total_seconds()
+
+        self.assertLess(request_time, MAX_RESPONSE_TIME)
+
+    def test_response_time_sync_action(self):
```

Comment:
Please remove this test and the other test in which you wait for a timeout. The timeout is already tested and by having these tests, you are adding 20 seconds to our tests! That's a 50% increate on my machine.

![screen shot 2017-01-27 at 11 58 56 pm](https://cloud.githubusercontent.com/assets/1444314/22395255/ce545c6c-e4ec-11e6-9d66-2878d72227d5.png)